src/Bicep.Core/Workspaces/SourceFileGrouping.cs

```diff
@@ -26,35 +26,31 @@ public record SourceFileGrouping(
         IFileResolver FileResolver,
         Uri EntryFileUri,
         ImmutableDictionary<Uri, FileResolutionResult> FileResultByUri,
-        ImmutableDictionary<StatementSyntax, UriResolutionResult> UriResultByModule,
+        ImmutableDictionary<ISourceFile, ImmutableDictionary<StatementSyntax, UriResolutionResult>> UriResultByModule,
```

Comment:
Out of interest, what was the motivation for this change?

Reply:
I need to be able to report in which file a `ModuleDeclarationSyntax` occurred when reporting that it requires restoration in `SourceFileGrouping::GetModulesToRestore`. The module dispatcher will use the source file to resolve any registry aliases used in the module path.